src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java

```diff
@@ -461,28 +459,52 @@ private EntityPlayerMP parseAnyPlayerValue(@NotNull final MinecraftServer server
     private IColony parseColonyValue(@NotNull final ICommandSender sender, @NotNull final MineColonyDataProvider mineColonyDataProvider, final String potentialArgumentValue)
     {
         final List<String> colonyNumberStrings = getColonyIdStrings(mineColonyDataProvider);
-        final Integer result = Ints.tryParse(potentialArgumentValue);
-        if (null != result)
-        {
-            int colonyNumber = result.intValue();
-            if (sender instanceof EntityPlayer && colonyNumber == -1)
-            {
-                final IColony icolony = mineColonyDataProvider.getIColonyByOwner(sender.getEntityWorld(), (EntityPlayer) sender);
-                if (icolony != null)
-                {
-                    colonyNumber = icolony.getID();
-                }
-            }
-            if (colonyNumberStrings.contains(String.valueOf(colonyNumber)))
-            {
-                return mineColonyDataProvider.getColony(colonyNumber, sender.getEntityWorld().provider.getDimension());
-            }
-            return null;
-        }
-        else
-        {
-            return null;
-        }
+        int dimensionId;
+        int colonyNumber;
+        if (potentialArgumentValue.contains("|"))
+        {
+        	String[] split = potentialArgumentValue.split(Pattern.quote("|"));
+        	dimensionId = Ints.tryParse(split[0]).intValue();
+        	colonyNumber = Ints.tryParse(split[1]).intValue();
+        }
+        else if (sender instanceof EntityPlayer)
+        {
+        	final Integer result = Ints.tryParse(potentialArgumentValue);
+        	if (null != result) {
+        		dimensionId = ((EntityPlayer) sender).dimension;
+        		colonyNumber = result.intValue();
+        	}
+        	else return null;
+        }
+        else 
+        {
+        	return null;
+        }
+    	if (sender instanceof EntityPlayer && colonyNumber == -1) 
```

Comment:
formatting is off here

Reply:
Right, I was trying to fix that by pressing ctrl+shift+f in eclipse but apparently that uses a different indenting style.

Reply:
There is a style file in our repo you can import

Reply:
Weird, though, it doesn't seem to show that in eclipse 🤔

Reply:
![image](https://user-images.githubusercontent.com/6438347/70866950-85262080-1f4e-11ea-9ce5-6cd85915f05f.png)

Reply:
you find it under documentation>eclipse

Reply:
The checkstyle thing?

Reply:
As far as I know you should be able to import the intellij .xml file into eclipse too

Reply:
If I use that, Ctrl+shift+f still changes parts of the code I didn't even touch

Reply:
> As far as I know you should be able to import the intellij .xml file into eclipse too

how? I usually just go with the default eclipse formatting

Reply:
You can just mark the code you want to format, and then apply formatting to it.

Reply:
still does bracing style differently than everywhere in the file

Reply:
In any case, my eclipse doesn't even show the indenting issues to me, weirdly

Reply:
for me it looks fine